src/components/MultiHopTrade/hooks/useReceiveAddress.tsx

```diff
@@ -24,7 +24,12 @@ export const getReceiveAddress = async ({
   const chainAdapter = getChainAdapterManager().get(chainId)
   if (!(chainAdapter && wallet)) return
   const { accountNumber } = bip44Params
-  const address = await chainAdapter.getAddress({ wallet, accountNumber, accountType, pubKey })
+  const address = await chainAdapter.getAddress({
+    wallet,
+    accountNumber,
+    accountType,
+    pubKey: pubKey!,
```

Comment:
Can we `if (!pubKey) return` above instead of non-null asserting?

Reply:
See https://github.com/shapeshift/web/pull/7655#discussion_r1743257622 - same rationale, undefined `pubKey` is actually fine and expected here for non-Ledger wallets (we only want to skip on-device derivation for Ledger, hence the maybe undefined `pubKey`, but we also just happen to not need open app checks here.

@0xApotheosis in addition to a comment, WDYT about opt-chaining instead of casting? This will produce the same effect - it will be falsey, meaning chain-adapters will *skip* *skipping* (double skip intended!) on-device derivation, and that may look a bit less ugly.

Reply:
Disregard the above, this guy was the goer https://github.com/shapeshift/web/pull/7655#discussion_r1743143840: e08c8e4ce8